<launch>	
	<node name="slam" pkg="slam" type="hitlc_slam_node" output="screen">
		
		<param name="save_path" value="/home/pharos/slam_ws/data/Kaist/" />
		
		<param name="g2o_solver" value="gn_var" />

		<param name="points_topic_name" value="/velodyne_points" />
		<param name="vehicle_state_topic_name" value="/vehicle/state2016" />
		<param name="gps_state_topic_name" value="/odom/novatel" />
		<param name="imu_topic_name" value="/ekf_imu" />

		<param name="use_vehicle_statas" value="true" />
		<param name="use_imu" value="true" />		
		<param name="use_ekf" value="true" />
		<param name="use_gps_init" value="true" />
		<param name="save_data" value="false" />

		<param name="min_scan_range" value="5.0" />
		<param name="max_scan_range" value="200.0" />
		
		<param name="ndt_res" value="1.0" />
		<param name="max_iter" value="200" />
		<param name="step_size" value="0.1" />
		<param name="trans_eps" value="0.01" />
		<param name="voxel_leaf_size" value="0.5" />

		<param name="min_add_scan_shift" value="1.0" />
		<param name="target_map_lengh" value="8.0" />
		<param name="max_lookup_LCdist" value="10.0" />
		<param name="max_score_for_LC" value="3.5" />
		<param name="min_edges_for_LC" value="2" />

		<param name="tf_x" value="0.0" />
		<param name="tf_y" value="0.0" />
		<param name="tf_z" value="0.0" />
		<param name="tf_roll" value="0.0" />
		<param name="tf_pitch" value="0.0" />
		<!-- <param name="tf_yaw" value="-0.23" /> -->
		<param name="tf_yaw" value="0.0" />

	</node>

	<node type="rviz" name="rviz" pkg="rviz" args="-d $(find slam)/config/ndt_slam.rviz" />
</launch>
